frontend/src/stores/settings.ts

```diff
@@ -0,0 +1,25 @@
+import { defineStore } from 'pinia';
+
+import { Settings, WalletType } from '@/types/settings';
+
+export const useSettings = defineStore('settings', {
+  state: () => ({
+    settings: undefined as Settings,
+  }),
+  actions: {
+    setConnectedWallet(connectedWallet: WalletType) {
+      this.settings = { ...this.settings, connectedWallet };
```

Comment:
If I'm not mistaken this is not necessary. Pinia is reactive by default and it doesn't require to replace the whole property. In turn, I think you can get rid of the whole action and just set the property directly where needed. That is how Pinia is meant to work.